tests/otel_integrations/test_httpx.py

```diff
@@ -247,6 +248,117 @@ async def test_async_httpx_client_instrumentation_with_capture_headers(
+    assert span['attributes'] == IsDict(expected_attributes).settings(partial=True)
+
+
+@pytest.mark.parametrize(*CAPTURE_JSON_BODY_PARAMETERS)
+async def test_async_httpx_client_instrumentation_with_capture_json_body(
+    exporter: TestExporter, content_type: str, body: Any, expected_attributes: dict[str, Any]
+):
+    with logfire.span('test span') as span:
+        assert span.context
+        trace_id = span.context.trace_id
+        async with httpx.AsyncClient(transport=create_transport()) as client:
+            logfire.instrument_httpx(client, capture_request_json_body=True)
+            try:
+                response = await client.post(
+                    'https://example.org/', headers={'Content-Type': content_type}, content=body
+                )
+            finally:
+                HTTPXClientInstrumentor().uninstrument()
+            traceparent_header = response.headers['traceparent']
+            assert f'{trace_id:032x}' == traceparent_header.split('-')[1]
+
+    span = exporter.exported_spans_as_dict()[0]
+    assert span['attributes'] == IsDict(expected_attributes).settings(partial=True)
+
+
+CAPTURE_FULL_REQUEST_ATTRIBUTES = {
+    'http.request.header.host',
+    'http.request.header.accept',
+    'http.request.header.accept-encoding',
+    'http.request.header.connection',
+    'http.request.header.user-agent',
+    'http.request.header.content-type',
+    'http.request.header.content-length',
+    'http.request.body.json',
+}
+
+
+def test_httpx_client_capture_full_request(exporter: TestExporter):
+    with logfire.span('test span') as span:
+        assert span.context
+        trace_id = span.context.trace_id
+        with httpx.Client(transport=create_transport()) as client:
+            logfire.instrument_httpx(client, capture_request_headers=True, capture_request_json_body=True)
+            try:
+                response = client.post(
+                    'https://example.org/', headers={'Content-Type': 'application/json'}, json={'hello': 'world'}
+                )
+            finally:
+                HTTPXClientInstrumentor().uninstrument()
+            traceparent_header = response.headers['traceparent']
+            assert f'{trace_id:032x}' == traceparent_header.split('-')[1]
+
+    span = exporter.exported_spans_as_dict()[0]
+    assert all(key in span['attributes'] for key in CAPTURE_FULL_REQUEST_ATTRIBUTES)
+
+
+async def test_async_httpx_client_capture_full_request(exporter: TestExporter):
+    with logfire.span('test span') as span:
+        assert span.context
+        trace_id = span.context.trace_id
+        async with httpx.AsyncClient(transport=create_transport()) as client:
+            logfire.instrument_httpx(client, capture_request_headers=True, capture_request_json_body=True)
+            try:
+                response = await client.post(
+                    'https://example.org/', headers={'Content-Type': 'application/json'}, json={'hello': 'world'}
```

Comment:
That's just because the content-type should be implied by the json kwarg. But we also need to test the content type actually missing. Add `content=b'{"hello": "world"}'` to one of the older tests without setting a content type to ensure that it doesn't capture anything.

Reply:
I wanted to test with other content types, as the parameterize shows.

Reply:
I read this wrongly, sorry. Fixed.